This patch adds a clock in the most left of the bar.

diff --git a/dwm.c b/dwm.c
index 4465af1..a394159 100644
--- a/dwm.c
+++ b/dwm.c
@@ -27,5 +27,6 @@ add time.h
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <time.h>
 #include <unistd.h>
 #include <sys/types.h>
@@ -172,5 +173,6 @@ declare gettime
 static long getstate(Window w);
 static unsigned int getsystraywidth();
 static int gettextprop(Window w, Atom atom, char *text, unsigned int size);
+static char* gettime(const char *format);
 static void grabbuttons(Client *c, int focused);
 static void grabkeys(void);
@@ -432,9 +432,13 @@ change the click positions
 		selmon = m;
 		focus(NULL);
 	}
 	if (ev->window == selmon->barwin) {
-		i = x = 0;
+		i = 0;
+		if(enable_clock)
+			x = TEXTW(gettime(clock_format));
+		else
+			x = 0;
 		do
 			x += TEXTW(tags[i]);
 		while (ev->x >= x && ++i < LENGTH(tags));
 		if (i < LENGTH(tags)) {
@@ -869,7 +876,11 @@ add to bar
		if (c->isurgent)
			urg |= c->tags;
	}
-	x = 0;
+	if(enable_clock){
+		char *clock_text = gettime(clock_format);
+		x = TEXTW(clock_text);
+		drw_text(drw, 0, 0, x, bh, lrpad/2, clock_text, 0);
+	} else { x = 0; }
	for (i = 0; i < LENGTH(tags); i++) {
		w = TEXTW(tags[i]);
		drw_setscheme(drw, scheme[m->tagset[m->seltags] & 1 << i ? SchemeTagSel : SchemeTag]);
@@ -1095,8 +1096,23 @@ function gettime
 	text[size - 1] = '\0';
 	XFree(name.value);
 	return 1;
 }
 
+char*
+gettime(const char *format)
+{
+	static char buffer[64];  // Static buffer for formatted time string
+	time_t raw_time;
+	struct tm* time_info;
+	time(&raw_time); // Get current time
+	time_info = localtime(&raw_time);  // Convert to local time
+
+	if (strftime(buffer, sizeof(buffer), format, time_info) == 0) {
+		return NULL;  // Formatting failed
+	}
+	return buffer;
+}
+ 
 void
 grabbuttons(Client *c, int focused)
 {
